src/github.com/matrix-org/dendrite/roomserver/storage/sql.go

```diff
@@ -326,20 +363,46 @@ func (s *statements) prepareEvents(db *sql.DB) (err error) {
 	if s.insertEventStmt, err = db.Prepare(insertEventSQL); err != nil {
 		return
 	}
+	if s.selectStateEventsByIDStmt, err = db.Prepare(selectStateEventsByIDSQL); err != nil {
+		return
+	}
 	return
 }
 
 func (s *statements) insertEvent(
 	roomNID, eventTypeNID, eventStateKeyNID int64,
 	eventID string,
 	referenceSHA256 []byte,
+	authEventNIDs []int64,
```

Comment:
What do you think about having a type alias of:
```go
type NID int64
```

Do you think this would be useful and clearer to the reader? That means you could drop all the `NID` suffixes on your variable names.